lib/Elastica/Reindex.php

```diff
@@ -0,0 +1,139 @@
+<?php
+namespace Elastica;
+
+use Elastica\Query\AbstractQuery;
+
+class Reindex
+{
+    const VERSION_TYPE = 'version_type';
+    const VERSION_TYPE_INTERNAL = 'internal';
+    const VERSION_TYPE_EXTERNAL = 'external';
+    const OPERATION_TYPE = 'op_type';
+    const OPERATION_TYPE_CREATE = 'create';
+    const CONFLICTS = 'conflicts';
+    const CONFLICTS_PROCEED = 'proceed';
+    const TYPE = 'type';
+    const SIZE = 'size';
+    const QUERY = 'query';
+
+    /**
+     * @var Index
+     */
+    protected $_oldIndex;
+
+    /**
+     * @var Index
+     */
+    protected $_newIndex;
+
+    /**
+     * @var array
+     */
+    private $_options;
+
+    public function __construct(Index $oldIndex, Index $newIndex, array $options = [])
+    {
+        $this->_oldIndex = $oldIndex;
+        $this->_newIndex = $newIndex;
+        $this->_options = $options;
+    }
+
+    public function run()
+    {
+        $body = $this->getBody($this->_oldIndex, $this->_newIndex, $this->_options);
+
+        $reindexEndpoint = new \Elasticsearch\Endpoints\Reindex();
+        $reindexEndpoint->setBody($body);
+
+        $this->_oldIndex->getClient()->requestEndpoint($reindexEndpoint);
+        $this->_newIndex->refresh();
+
+        return $this->_newIndex;
+    }
+
+    protected function getBody($oldIndex, $newIndex, $options)
+    {
+        $body = array_merge([
+            'source' => $this->getSourcePartBody($oldIndex, $options),
+            'dest' => $this->getDestPartBody($newIndex, $options)
+        ], self::resolveBodyOptions($options));
+
+        return $body;
+    }
+
+    private function getSourcePartBody(Index $index, array $options)
+    {
+        $sourceBody = array_merge([
+            'index' => $index->getName(),
+        ], $this->resolveSourceOptions($options));
+
+        $sourceBody = $this->setSourceQuery($sourceBody);
+        $sourceBody = $this->setSourceType($sourceBody);
+
+        return $sourceBody;
+    }
+
+    private function getDestPartBody(Index $index, array $options)
+    {
+        return array_merge([
+            'index' => $index->getName(),
+        ], $this->resolveDestOptions($options));
+    }
+
+    private static function resolveSourceOptions(array $options)
```

Comment:
This should also be not static and start with _* Also make it protected if there is no need to make it private.